src/auth/service/roles/roles.go

```diff
@@ -0,0 +1,51 @@
+package roles
+
+import (
+	"context"
+	"github.com/consensys/quorum-key-manager/pkg/errors"
+	"github.com/consensys/quorum-key-manager/src/auth/entities"
+	"sync"
+
+	"github.com/consensys/quorum-key-manager/src/auth"
+	"github.com/consensys/quorum-key-manager/src/infra/log"
+)
+
+type Interactor struct {
+	mux    sync.RWMutex
+	roles  map[string]*entities.Role
+	logger log.Logger
+}
+
+var _ auth.Roles = &Interactor{}
+
+func New(logger log.Logger) *Interactor {
```

Comment:
I would rather have a single file with all its methods together since there is not so much complexity on the methods itself for separation